src/sentry/api/event_search.py

```diff
@@ -1056,63 +1104,84 @@ def to_list(value):
 
 
 def format_search_filter(term, params):
-    project_to_filter = None  # Used to avoid doing multiple conditions on project ID
+    projects_to_filter = []  # Used to avoid doing multiple conditions on project ID
     conditions = []
     group_ids = None
     name = term.key.name
     value = term.value.value
     if name in (PROJECT_ALIAS, PROJECT_NAME_ALIAS):
         if term.operator == "=" and value == "":
             raise InvalidSearchQuery("Invalid query for 'has' search: 'project' cannot be empty.")
-        project = None
-        try:
-            project = Project.objects.get(id__in=params.get("project_id", []), slug=value)
-        except Exception as e:
-            if not isinstance(e, Project.DoesNotExist) or term.operator != "!=":
-                raise InvalidSearchQuery(
-                    f"Invalid query. Project {value} does not exist or is not an actively selected project."
-                )
-        else:
+        slugs = to_list(value)
+        projects = {
+            p.slug: p.id
+            for p in Project.objects.filter(id__in=params.get("project_id", []), slug__in=slugs)
+        }
+        missing = [slug for slug in slugs if slug not in projects]
+        if missing and term.operator not in ("!=", "NOT IN"):
+            raise InvalidSearchQuery(
+                f"Invalid query. Projects {missing} do not exist or are not actively selected."
+            )
+        project_ids = list(sorted(projects.values()))
```

Comment:
Could/should we do this with a `.sort()` on the query instead?

Reply:
SInce we put it into the dict after load we'd lose any sort order from the query.